admin/class-yoast-notification.php

```diff
@@ -52,6 +60,165 @@ public function get_id() {
 	}
 
 	/**
+	 * Retrieve nonce identifier.
+	 *
+	 * @return null|string Nonce for this Notification.
+	 */
+	public function get_nonce() {
+		if ( $this->options['id'] && empty( $this->options['nonce'] ) ) {
+			$this->options['nonce'] = wp_create_nonce( $this->options['id'] );
+		}
+
+		return $this->options['nonce'];
+	}
+
+	/**
+	 * Get the type of the notification
+	 *
+	 * @return string
+	 */
+	public function get_type() {
+		return $this->options['type'];
+	}
+
+	/**
+	 * Priority of the notification
+	 *
+	 * Relative to the type
+	 *
+	 * @return float 0-1
+	 */
+	public function get_priority() {
+		return $this->options['priority'];
+	}
+
+	/**
+	 * Get the User Meta key to check for dismissal of notification
+	 *
+	 * @return string User Meta Option key that registers dismissal.
+	 */
+	public function get_dismissal_key() {
+		if ( empty( $this->options['dismissal_key'] ) ) {
+			return $this->options['id'];
+		}
+
+		return $this->options['dismissal_key'];
+	}
+
+	/**
+	 * Is this Notification persistent
+	 *
+	 * @return bool True if persistent, False if fire and forget.
+	 */
+	public function is_persistent() {
+		$id = $this->get_id();
+
+		return ! empty( $id );
+	}
+
+	/**
+	 * Check if the notification is relevant for the current user
+	 *
+	 * @return bool True if a user needs to see this Notification, False if not.
+	 */
+	public function display_for_current_user() {
+		// If the notification is for the current page only, always show.
+		if ( ! $this->is_persistent() ) {
```

Comment:
Should a non-persistent notice be a `Yoast_Alert`? Then `Yoast_Alert` and `Yoast_Notification` can both implement the `Yoast_Notification_Interface`.